### PR TITLE
Button refactor

### PR DESCRIPTION
**Pull request by RCopeland**

### CODE REVIEW

File: src/angular/src/app/spark-core-angular/components/sprk-ordered-list/sprk-ordered-list.component.spec.ts [hunk 1]
```diff
@@ -28,6 +28,18 @@ describe('SparkOrderedListComponent', () => {
     expect(element.classList.toString()).toEqual(component.getClasses());
   });
 
+  it('should add the correct classes if bare is set as listType', () => {
```
Comment (afebbraro) [inline src/angular/src/app/spark-core-angular/components/sprk-ordered-list/sprk-ordered-list.component.spec.ts]: Did we miss some list tests or is this related to buttons?

File: src/angular/src/app/spark-core-angular/directives/inputs/sprk-button/sprk-button.directive.spec.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }));
+
+  it('should create itself', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should add classes if additionalClasses has a value', () => {
+    fixture.detectChanges();
+    expect(buttonElement.classList.toString()).toEqual(
+      'sprk-c-Button sprk-u-man'
```
Comment (afebbraro) [inline src/angular/src/app/spark-core-angular/directives/inputs/sprk-button/sprk-button.directive.spec.ts]: Where is the setting of additionalClasses to "sprk-u-man"?

File: src/angular/src/app/spark-core-angular/components/sprk-ordered-list/sprk-ordered-list.component.spec.ts (hunk 1 above)
Comment (RCopeland) [reply src/angular/src/app/spark-core-angular/components/sprk-ordered-list/sprk-ordered-list.component.spec.ts]: missed a few lines to test

File: src/angular/src/app/spark-core-angular/directives/inputs/sprk-button/sprk-button.directive.spec.ts (hunk 2 above)
Comment (RCopeland) [reply src/angular/src/app/spark-core-angular/directives/inputs/sprk-button/sprk-button.directive.spec.ts]: its up in the declaration
